test.js

```diff
@@ -284,3 +355,42 @@ test('iterators', function (t) {
 
   t.ok(require('./').DeferredIterator)
 })
+
+test('iterator - direct operations', function (t) {
+  t.plan(6)
+
+  var db = {
+    iterator: function (options) {
+      return {
+        next: function (cb) {
+          cb(null, 'key', 'value')
+        },
+        end: function (cb) {
+          process.nextTick(cb)
+        }
+      }
+    },
+    open: function (options, callback) {
+      process.nextTick(callback)
+    }
+  }
+  var ld = new DeferredLevelDOWN(db)
+  var it = ld.iterator()
+
+  ld.open(function (err) {
+    t.error(err, 'no error')
+    var nextFirst = false
+
+    it.next(function (err, key, value) {
+      nextFirst = true
+      t.error(err, 'no error')
+      t.equal(key, 'key')
+      t.equal(value, 'value')
+    })
+
+    it.end(function (err) {
```

Comment:
Why does this test call `it.end` immediately after `it.next`? Does `deferred-leveldown` do something special in this case?

Reply:
Good question. I stole the code from the other test and tweaked it. It makes more sense to have them called immediately in the deferred case and instead we should probably do `it.end()` after `it.next()` has finished here. Or we can probably remove it completely even.